rust/worker/src/segment/record_segment.rs

```diff
@@ -599,6 +599,22 @@ impl RecordSegmentReader<'_> {
         self.id_to_data.get("", offset_id).await
     }
 
+    pub(crate) async fn data_exists_for_user_id(
```

Comment:
Is this a workaround for the fact that we don't propagate NotFound up? If so can we mark a todo here for that?

Reply:
In general, I feel like the blockfilereader should provide a `has()` or a `contains()` equivalent API since it is a common workflow to check for the presence/absence of something in the storage. So, I am generally of the view point that we should not use the `get()` API and handle errors to implement a `has()`

Reply:
sure agreed